instrumentor/controllers/sourceinstrumentation/common.go

```diff
@@ -0,0 +1,329 @@
+}
+
+func uninstrumentWorkload(
+	ctx context.Context,
+	k8sClient client.Client,
+	objKind k8sconsts.WorkloadKind,
+	key client.ObjectKey,
+	scheme *runtime.Scheme) (ctrl.Result, error) {
+	obj := workload.ClientObjectFromWorkloadKind(objKind)
+	err := k8sClient.Get(ctx, key, obj)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	instrumented, _, err := sourceutils.IsObjectInstrumentedBySource(ctx, k8sClient, obj)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !instrumented {
+		err = errors.Join(err, deleteWorkloadInstrumentationConfig(ctx, k8sClient, obj))
+		err = errors.Join(err, removeReportedNameAnnotation(ctx, k8sClient, obj))
+	}
+	return ctrl.Result{}, err
+}
+
+func instrumentWorkload(
+	ctx context.Context,
+	k8sClient client.Client,
+	objKind k8sconsts.WorkloadKind,
+	key client.ObjectKey,
+	scheme *runtime.Scheme) (ctrl.Result, error) {
+
+	logger := log.FromContext(ctx)
+
+	obj := workload.ClientObjectFromWorkloadKind(objKind)
+	err := k8sClient.Get(ctx, key, obj)
+	if err != nil {
+		// Deleted objects should be filtered in the event filter
+		return ctrl.Result{}, err
+	}
+
+	workloadObj, err := workload.ObjectToWorkload(obj)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	enabled, markedForInstrumentationCondition, err := sourceutils.IsObjectInstrumentedBySource(ctx, k8sClient, obj)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if !enabled {
+		return ctrl.Result{}, nil
+	}
+
+	instConfigName := workload.CalculateWorkloadRuntimeObjectName(key.Name, objKind)
+	ic, err := requestOdigletsToCalculateRuntimeDetails(ctx, k8sClient, instConfigName, key.Namespace, obj, scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if ic == nil {
+		ic = &v1alpha1.InstrumentationConfig{}
+		err = k8sClient.Get(ctx, types.NamespacedName{Name: instConfigName, Namespace: key.Namespace}, ic)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	markedForInstChanged := meta.SetStatusCondition(&ic.Status.Conditions, markedForInstrumentationCondition)
+	runtimeDetailsChanged := initiateRuntimeDetailsConditionIfMissing(ic, workloadObj)
+	agentEnabledChanged := initiateAgentEnabledConditionIfMissing(ic)
+
+	if markedForInstChanged || runtimeDetailsChanged || agentEnabledChanged {
+		logger.Info("Updating initial instrumentation status condition of InstrumentationConfig", "name", instConfigName, "namespace", key.Namespace)
+		ic.Status.Conditions = sortIcConditionsByLogicalOrder(ic.Status.Conditions)
+
+		err = k8sClient.Status().Update(ctx, ic)
+		if err != nil {
+			logger.Info("Failed to update status conditions of InstrumentationConfig", "name", instConfigName, "namespace", key.Namespace)
+			return k8sutils.K8SUpdateErrorHandler(err)
+		}
+	}
+
+	return ctrl.Result{}, err
+}
+
+func requestOdigletsToCalculateRuntimeDetails(ctx context.Context, k8sClient client.Client, instConfigName string, namespace string, obj client.Object, scheme *runtime.Scheme) (*v1alpha1.InstrumentationConfig, error) {
```

Comment:
nit: originally, this controller would actually set a flag named `invalidateRuntimeDetection` which the odiglet react to.

I think the function name can be changed now that the flow no longer works like this.

How about 

```suggestion
func createInstrumentationConfigForWorkload(ctx context.Context, k8sClient client.Client, instConfigName string, namespace string, obj client.Object, scheme *runtime.Scheme) (*v1alpha1.InstrumentationConfig, error) {
```

Reply:
sgtm